shell/platform/android/io/flutter/embedding/engine/FlutterEngine.java

```diff
@@ -183,8 +184,14 @@ public FlutterEngine(
       boolean automaticallyRegisterPlugins
   ) {
     this.flutterJNI = flutterJNI;
+
+    // Add the --write-service-info flag to the incoming Dart VM args.
+    dartVmArgs = dartVmArgs == null ? new String[] {} : dartVmArgs;
```

Comment:
Should this knowledge be in the Flutter engine? The FlutterLoader seems to be a better encapsulation for knowledge like these since all other vm args are added there.

Reply:
Including constant hard-coded ones like "--icu-symbol-prefix=_binary_icudtl_dat"

Reply:
Makes sense. I'll move it over there.